torch/csrc/autograd/python_variable.cpp

```diff
@@ -221,7 +223,7 @@ static PyObject* THPVariable_make_subclass(PyObject* _ignored, PyObject* args, P
   // ```
   data.unsafeGetTensorImpl()->set_allow_tensor_metadata_change(true);
   auto var = data.set_requires_grad(r.toBool(2));
-  return THPVariable_NewWithVar((PyTypeObject*)cls, std::move(var));
+  return THPVariable_NewWithVar((PyTypeObject*)cls, std::move(var), c10::impl::PyInterpreterStatus::DEFINITELY_UNINITIALIZED);
```

Comment:
i don't understand tensor well enough to know why this is safe and _pynew below is not.  what is var?

Reply:
var is data; the old code is written in a needlessly oblique way. Data is fresh because it is detached `auto data = r.tensor(1).detach();`.

`_pynew` is not OK because if you pass it an integer address it will return a preexisting Tensor object as is, so it could have come from another interpreter in that case.